### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Unit tests added

### CODE REVIEW

File: pkg/cloud/mocks/mock_cloud.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
+// Code generated by MockGen. DO NOT EDIT.
```
Comment (mkumatag) [inline pkg/cloud/mocks/mock_cloud.go:1]: how is this mock generated and same comment applies for the mock_mounter.go client as well.

**Review by mkumatag — CHANGES_REQUESTED** [review]

File: pkg/cloud/mocks/mock_cloud.go (hunk 1 above)
Comment (Madhan-SWE) [reply pkg/cloud/mocks/mock_cloud.go:1]: In the next PRs,
We will be adding commands to generate mocks from Make file: https://github.com/Madhan-SWE/ibm-powervs-block-csi-driver/blob/998925e00bae30813981c687561b3a33df2d5cbb/Makefile#L66

Commands to generate mock: https://github.com/Madhan-SWE/ibm-powervs-block-csi-driver/blob/feature_remove_pvm_instance_id_v3/hack/update-gomock


Should we also mention in the generated mocks about how we generated those?

Comment (mkumatag) [reply pkg/cloud/mocks/mock_cloud.go:1]: They are all autogenerated files, hence you hardly have any control. lgtm

**Review by mkumatag — APPROVED** [review]
/lgtm
/approve

**Pull request merged**